assets/data0_21pure/ui/porkui/blocklist.rml

```diff
@@ -0,0 +1,192 @@
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+
+-->
+
+<rml>
+<head>
+	<title>players</title>
+	<link type="text/template" href="template_ingame.rml" />
+	<link type="text/css" rel="stylesheet" href="css/blocklist.rcss" />
+	<!-- <script src="as/players.as" /> -->
+	<script>
+		Element @playersDatagrid;
+		Element @blockedDatagrid;
+		ElementForm @playerForm;
+		ElementFormControl @playerName;
+		ElementFormControl @playerNumArgs;
+		Element @playerOptions;
+		Element @playerText;
+		ElementFormControl @playerValue;
+
+		Element @blockBtn;
+		Element @viewprofileBtn;
+
+		String @selsteamid;
+		String @selname;
+
+		bool unblock = false;
+
+
+		void onPlayersLoad( Element @body, Event @evt )
+		{
+			onTemplateLoad( body, evt );
+
+			@playersDatagrid = body.getElementById( 'players_datagrid' );
+			@blockedDatagrid = body.getElementById( 'blocked_datagrid' );
+			@playerForm = body.getElementById('players_form_start');
+			@playerName = body.getElementById( "player_name" );
+			@playerNumArgs = body.getElementById( "player_numargs" );
+			@playerOptions = body.getElementById( "player_options" );
+			@playerText = body.getElementById( "player_text" );
+			@blockBtn = body.getElementById( "block_btn" );
+			@viewprofileBtn = body.getElementById( "viewprofile_btn" );
+
+		}
+
+		void onPlayersClick( Element @self, Event @evt )
+		{
+			window.close();
+		}
+
+		void onPlayersShow( Element @body, Event @evt )
+		{
+			blockBtn.css( 'visibility', 'hidden' );
+			viewprofileBtn.css( 'visibility', 'hidden' );
+			animationsOnShow();
+
+			playersSetDataSources();
+		}
+
+		void onPlayersHide( Element @body, Event @evt )
+		{
+			// force datasource update on next show event
+			playersClearDataSources();
+		}
+
+		void playersSetDataSources( void )
+		{
+			cast<ElementDataGrid>(blockedDatagrid).setDataSource( 'blockedplayers.what' );
+			cast<ElementDataGrid>(playersDatagrid).setDataSource( 'gameajax.players' );
+		}
+
+		void playersClearDataSources( void )
+		{		
+			cast<ElementDataGrid>(playersDatagrid).setDataSource( '' );
+			cast<ElementDataGrid>(blockedDatagrid).setDataSource( '' );
+		}
+
+		void onPlayersRowSelect( Element @elem, Event @ev )
+		{
+			unblock = false;
+			int selectedRow = ev.getParameter( 'index', -1 );			
+			if( selectedRow < 0 ) {
+				return;
+			}
+
+			DataSource @data = getDataSource( 'gameajax' );
+
+			// grab information for the picked row from the table
+			String name = data.getField( 'players', selectedRow, 'name' );
+			int playerid = data.getField( 'players', selectedRow, 'value' ).toInt();
+			String steamid = data.getField( 'players', selectedRow, 'steamid' );
+
+
+			blockBtn.css( 'display', 'inline-block' );
+			blockBtn.css( 'visibility', 'visible' );
+
+			viewprofileBtn.css( 'display', 'inline-block' );
+			viewprofileBtn.css( 'visibility', 'visible' );
+			blockBtn.setInnerRML("Block " + name);
```

Comment:
All this cgame steam filtering does is ignore chats from users, right?
I am not sure if blocking is the correct term for this.